shared/constants/chat2/message.js

```diff
@@ -633,15 +647,29 @@ const validUIMessagetoMessage = (
 
   switch (m.messageBody.messageType) {
     case RPCChatTypes.commonMessageType.text:
-      const rawText: string = m.messageBody.text?.body ?? ''
+      const messageText = m.messageBody.text
+      const rawText: string = messageText?.body ?? ''
+      const payments = messageText?.payments ?? null
       return makeMessageText({
         ...common,
         ...explodable,
         decoratedText: m.decoratedTextBody ? new HiddenString(m.decoratedTextBody) : null,
         hasBeenEdited: m.superseded,
-        inlinePaymentIDs: m.paymentInfos
```

Comment:
`m.paymentInfos` is empty on first thread load after service restart, `messageBody.text.payments` is more reliably filled.